src/PlaybackLooping.h

```diff
@@ -0,0 +1,101 @@
+/*!********************************************************************
+*
+ Audacity: A Digital Audio Editor
+
+ PlaybackLooping.h
+
+ Anton Gerasimov
+
+ **********************************************************************/
+
+#pragma once
+
+#include <wx/dc.h>
+
+#include "ViewInfo.h"
+#include "TrackPanelCell.h"
+#include "UIHandle.h"
+
+//! Playback looping service (processor).
+class AUDACITY_DLL_API PlaybackLooping
+{
+public:
+    PlaybackLooping();
+
+    static PlaybackLooping& GetInstance();
+
+    bool mLoopingEnabled{ false };
+    bool IsLoopingEnabled() const;
+    void EnableLooping(bool enable);
+
+    enum class ProcessingStates
+    {
+        kIdle,
+        kEnterLooping,
+        kLooping,
+        kDragging
```

Comment:
The "state machine" of older AdornedRulerPanel code, left-over from before the reimplementation with CellularPanel -- is not a good model to follow.

You should not need this enumeration.  CellularPanel keeps track of state transitions between unclicked, clicked, dragging, button-up, and a UIHandle just needs to supply callbacks for those.